python/iceberg/api/schema.py

```diff
@@ -49,6 +49,8 @@ def __init__(self, *argv):
         self._lowercase_name_to_id = None
         self._id_to_name = None
 
+        self.lazy_id_to_name()
```

Comment:
Probably fine to keep this PR not too complex, but if we're no longer lazy loading a lot of the code in this class can be a bit more succinct.  I know this currently matches what's in java, but I just find it a little strange that the functions are named as lazy, but at least in the case of  lazy_id_to_name(), we're eagerly loading in the constructor.

Reply:
Yep, felt the same thing when I added it. We may update Java implementation as well.

Reply:
rather than trying to be 1:1 compliant w/ Java lets try and write idiomatic code. I agree with @TGooch44 that we could simplify this.

Reply:
Updated